src/lib/ndpi_main.c

```diff
@@ -4536,12 +4536,16 @@ int load_categories_file_fd(struct ndpi_detection_module_struct *ndpi_str,
       category = strtok_r(NULL, "\t", &saveptr);
 
       if(category) {
-	int rc = ndpi_load_category(ndpi_str, name,
-				    (ndpi_protocol_category_t) atoi(category),
-				    user_data);
+        cat_id = atoi(category);
```

Comment:
Hmmm... Personally, I'd replace such stuff with `strtonum`. Ofc, this is a BSD specific function, but I guess it's easy to write our own implementation from scratch.

@IvanNardi @utoni what do you think about it?